### PR TITLE
XW-4851 | [MercuryApi] Cleanup snippets for file usage list from broken html tags

### PR DESCRIPTION
**Pull request by rybmat**
https://wikia-inc.atlassian.net/browse/XW-4851

@Wikia/x-wing

### CODE REVIEW

File: extensions/wikia/MercuryApi/handlers/MercuryApiFilePageHandler.class.php [hunk 1]
```diff
@@ -12,8 +12,25 @@ public static function getFileContent( Title $title ) {
 		$details = WikiaFileHelper::getMediaDetail( $title );
 		$mediaObject = ArticleAsJson::createMediaObject( $details, $title->getText() );
 
+		// if article contains user provided HTML which is invalid (e.g. to many </div>), snippetter treat it as a text
```
Comment (hakubo) [inline extensions/wikia/MercuryApi/handlers/MercuryApiFilePageHandler.class.php]: 'too'

File: extensions/wikia/MercuryApi/handlers/MercuryApiFilePageHandler.class.php [hunk 2]
```diff
@@ -12,8 +12,25 @@ public static function getFileContent( Title $title ) {
...
+		// and it is not removed as other tags. In mobile-wiki we do not escape snippets, therefore invalid html may
+		// cause errors on front-end side
+		$fileUsageList = array_map( function( $item ) {
```
Comment (hakubo) [inline extensions/wikia/MercuryApi/handlers/MercuryApiFilePageHandler.class.php]: should we check if fileUsageList is an array?

Comment (rybmat) [reply extensions/wikia/MercuryApi/handlers/MercuryApiFilePageHandler.class.php]: we don't have to, $fileUsageData['fileList'] is always an array and array_map returns array if array provided as source

File: extensions/wikia/MercuryApi/handlers/MercuryApiFilePageHandler.class.php (hunk 1 above)
Comment (rybmat) [reply extensions/wikia/MercuryApi/handlers/MercuryApiFilePageHandler.class.php]: done line below

**Review by hakubo — APPROVED** [review]

File: extensions/wikia/MercuryApi/handlers/MercuryApiFilePageHandler.class.php (hunk 1 above)
Comment (KockaAdmiralac) [reply extensions/wikia/MercuryApi/handlers/MercuryApiFilePageHandler.class.php]: I think they meant `e.g. to many </div>` -> `e.g. too many </div>`

**Pull request merged**